src/lib.rs

```diff
@@ -430,6 +432,311 @@ impl<T> OptionExt<T> for Option<T> {
     }
 }
 
+/// Additions to [`Future`](futures::Future).
```

Comment:
To avoid having to put a bunch of stuff in `cfg` attributes, this code will likely want to be put into a module (something like `futures_crate`?) and then we can have a minimal number of `cfg`s: one for the crate, one for the module, and one for the re-export of the trait.

Reply:
> something like `futures_crate`

Perhaps just `futures01` and then the standard library can be `futures`.